siliconcompiler/constraints/asic_pins.py

```
@@ -289,56 +289,66 @@ def get_layer(self, step: Optional[str] = None, index: Optional[Union[str, int]]
+                edge = 2
+            elif edge in ("north", "top"): # Top edge in 2D
+                edge = 3
+            elif edge in ("west", "left"):
+                edge = 4
+            elif edge in ("up", "top-face"):
+                edge = 5
+            elif edge in ("down", "bottom-face"):
+                edge = 6
             else:
-                raise ValueError(f"{side} is a not a recognized side")
+                raise ValueError(f"{edge} is not a recognized edge")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)